packages/nextjs/mocks/msw/groq/create-groq-handler.ts

```diff
@@ -40,22 +52,22 @@ function parseParams(searchParams: object) {
   );
 }
 
-/**
- * Recursively finds all nested objects
- */
-function flattenObjects(obj: object, results = new Set<object>()) {
-  if (results.has(obj)) {
-    // We've already traversed this object
-    return results;
-  }
-
-  results.add(obj);
+const INEFFICIENT_QUERY_THRESHOLD = 5_000;
 
-  Object.values(obj).forEach((child) => {
-    if (child && typeof child === "object") {
-      flattenObjects(child, results);
-    }
-  });
-
-  return results;
+export async function executeQuery(dataset: Dataset, query: string, params: Record<string, string>): Promise<unknown> {
+  const parsed = groqJs.parse(query, { params });
+  const streamResult = await groqJs.evaluate(parsed, { dataset, params });
+  const start = Date.now();
+  const result = await streamResult.get();
+  const elapsed = Date.now() - start;
+  if (elapsed >= INEFFICIENT_QUERY_THRESHOLD) {
+    // Issue a warning!
+    console.warn(`
+      [groq-handler] WARNING: this query took ${elapsed} ms to mock execute.
+      This usually indicates an inefficient query, and you should consider improving it.
+      ${query.includes("&&") ? "Instead of using [a && b], consider using [a][b] instead!" : ""}
+      Inefficient query: \n${query}
+    `);
+  }
```

Comment:
Almost all of our queries took 15-20 seconds to execute.
By simply changing `[a && b]` to `[a][b]`, these queries went down to 1-3 seconds.

Reply:
Classic. I've found this to be the case even in production groq queries, so I now generally lean toward splitting up filters into something like `[a][b]` right from the get-go.